test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java

```diff
@@ -151,9 +152,16 @@ public boolean isShutdown()
         @Override
         public synchronized void startup()
         {
+            startup(AbstractCluster.this);
+        }
+
+        public synchronized void startup(ICluster cluster)
+        {
+            if (cluster != AbstractCluster.this)
+                throw new IllegalArgumentException("Only the owning cluster can be used for startup"); //TODO why have this in the API?
```

Comment:
I was bitten by this when doing restart since I kept passing in the cluster; I don't mind removing the API as its unsafe anyways.

Reply:
actually its too big a change.  The user API is the same as the plugin api (instance), so this leaks to user but is needed by plugin.

Reply:
Really, the only way to solve this is to split the user API from the implementation API.  Ill file a JIRA for this

Reply:
Created https://issues.apache.org/jira/browse/CASSANDRA-15602. This would also help with binary compatibility, if we make changes to the user API it wouldn't affect the plugin API so each branch doesn't need to be changed.